programs/mango-v4/src/health/cache.rs

```diff
@@ -230,14 +231,14 @@ impl TokenInfo {
     }
 }
 
-/// Information about reserved funds on Serum3 open orders accounts.
+/// Information about reserved funds on Serum3 and Openbook V2 open orders accounts.
 ///
 /// Note that all "free" funds on open orders accounts are added directly
 /// to the token info. This is only about dealing with the reserved funds
 /// that might end up as base OR quote tokens, depending on whether the
 /// open orders execute on not.
 #[derive(Clone, Debug)]
-pub struct Serum3Info {
+pub struct SpotInfo {
```

Comment:
Significant issue: `SpotInfo::market_index`: This will be 0 both for the 0th serum3 market and the 0th openbook market, causing issues for all functions that try to look up a SpotInfo based on a serum3 or openbook market index.

Propably store `serum3_index : Option<Serum3MarketIndex, obv2_index: Option<OpenBookV2MarketIndex>`? This would have been seen at compile time if these indexes weren't just u16...